modules/salesforce_pull/salesforce_pull.module

```diff
@@ -11,14 +11,91 @@
 define('SALESFORCE_PULL_QUEUE', 'salesforce_pull');
 
 /**
+ * Implements hook_menu().
+ */
+function salesforce_pull_menu() {
+  $items = array();
+
+  $items['salesforce/webhook/pull'] = array(
+    'title' => 'Salesforce pull webhook callback',
+    'page callback' => 'salesforce_pull_webhook',
+    'access callback' => 'salesforce_pull_webhook_access',
+    'type' => MENU_CALLBACK,
+  );
+
+  return $items;
+}
+
+/**
+ * Access callback for Salesforce webhooks.
+ */
+function salesforce_pull_webhook_access() {
+  if (variable_get('salesforce_pull_webhook_enable') && $_GET['key'] === variable_get('salesforce_pull_webhook_key')) {
+    return user_access('access content');
+  }
+
+  return FALSE;
+}
+
+/**
+ * Webhook callback for salesforce pull. Returns status of 200 for successful
+ * attempt or 403 for a failed pull attempt (SF not authorized, threshhold
+ * reached, etc.
+ */
+function salesforce_pull_webhook() {
+  if (salesforce_pull()) {
+    $code = '200';
+    // Queue is populated, but not processed yet so we manually do some of what
+    // drupal_cron_run() does to trigger processing of our pull queue.
+    $queues = salesforce_cron_queue_info();
+    $info = $queues[SALESFORCE_PULL_QUEUE];
+    $callback = $info['worker callback'];
+    $end = time() + (isset($info['time']) ? $info['time'] : 15);
+    $queue = DrupalQueue::get(SALESFORCE_PULL_QUEUE);
+    while (time() < $end && ($item = $queue->claimItem())) {
+      try {
+        call_user_func($callback, $item->data);
+        $queue->deleteItem($item);
+      }
+      catch (Exception $e) {
+        // In case of exception log it and leave the item in the queue
+        // to be processed again later.
+        watchdog_exception('salesforce_pull', $e);
+      }
+    }
+  }
+  else {
+    $code = '403';
+  }
+
+  http_response_code($code);
+  drupal_exit();
+}
+
+/**
  * Implements hook_cron().
  */
 function salesforce_pull_cron() {
+  salesforce_pull();
+}
+
+/**
+ * Callback for the standard pull process used by webhooks and cron.
+ */
+function salesforce_pull() {
   $sfapi = salesforce_get_api();
-  if ($sfapi->isAuthorized()) {
+  if ($sfapi->isAuthorized() && salesforce_pull_check_throttle()) {
     salesforce_pull_get_updated_records();
     salesforce_pull_process_deleted_records();
+
+    $request_time = $_SERVER['REQUEST_TIME_FLOAT'] ? $_SERVER['REQUEST_TIME_FLOAT'] : $_SERVER['REQUEST_TIME'];
+    variable_set('salesforce_pull_last_sync', $request_time);
```

Comment:
Agreed - a unified getter sounds like a good idea, especially if we're doing it inconsistently in other places. Since the throttle configuration is measured in seconds (i.e. you could set it to 1 second) IMO it makes way more sense to use `REQUEST_TIME_FLOAT` if it's available.